Compactable/separate

```diff
@@ -0,0 +1,49 @@
+{-
+
+mapEither maps a function to Either over the `f` and `separate`s `Left`s from `Right`s.
+
+Examples:
+
+```
+    let E = constructors (./Either/Type Natural Natural)
+
+in  ./Compactable/mapEither
```

Comment:
Same here. Obviously harder to avoid the name. I guess we need like dhall-doctest now, right 😁